src/app/view/components/dot-key-value/dot-key-value.component.html

```diff
@@ -0,0 +1,41 @@
+<p-table
+    #table
+    [value]="variablesBackup"
+    sortField="clazz"
+    (keydown.enter)="$event.stopPropagation()"
+    sortOrder="0"
+    dataKey="id"
+>
+    <ng-template pTemplate="header" let-columns>
+        <tr>
+            <th [ngStyle]="{ 'text-align': 'left', width: '33%' }">
+                {{ messages['keyValue.key_header.label'] }}
+            </th>
+            <th [ngStyle]="{ 'text-align': 'left', width: '33%' }">
+                {{ messages['keyValue.value_header.label'] }}
+            </th>
+            <th [ngStyle]="{ 'text-align': 'right', width: '185px', 'padding-right': '32px' }">
+                {{ messages['keyValue.actions_header.label'] }}
+            </th>
+        </tr>
+    </ng-template>
+    <ng-template pTemplate="body" let-variable let-rowIndex="rowIndex">
+        <dot-key-value-table-row
+            #tableRow
+            [variable]="variable"
+            [variableIndex]="rowIndex"
+            [variablesList]="variables"
+            (cancel)="onCancel($event)"
+            (delete)="deleteVariable($event)"
+            (save)="saveVariable($event)"
+        >
+        </dot-key-value-table-row>
+    </ng-template>
+    <ng-template pTemplate="emptymessage" let-variables>
+        <tr>
+            <td [attr.colspan]="3">
```

Comment:
`[attr.colspan]` is when you have to do some sort of dynamic value but templates takes regular HTML so...

```suggestion
            <td colspan="3">
```